langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/OllamaStreamingResponseBuilder.java

```diff
@@ -13,6 +16,7 @@ class OllamaStreamingResponseBuilder {
 
     private StringBuffer contentBuilder = new StringBuffer();
     private volatile TokenUsage tokenUsage;
+    private volatile List<ToolCall> toolsCall;
```

Comment:
Why not just store a reference (AtomicReference) to the last partial response object and generate Response on build() ?

Reply:
This implementation is strongly inspired by StreamingResponseBuilder and AzureOpenAiStreamingResponseBuilder.
The difference is that it seems Ollama always returns all tool calls in one response, unlike other implementations that return them in deltas.

My intention was to make it as similar as possible to other implementations.

Reply:
Ideally this code sohuld handle the case when Ollama will change the streaming behaviour. If could start returning token-by-token the same way as other LLM APIs do. Would be great to investigate why Ollama behaves this way, perharps there is more info on their github/discord/etc?